src/pkg/cloudclient/restapi/client.go

```diff
@@ -18,3 +19,16 @@ func NewClientFromToken(address string, token string) (*cloudapi.ClientWithRespo
 func IsErrorStatus(statusCode int) bool {
 	return statusCode < 200 || statusCode >= 300
 }
+
+func WithStatusCheck[T output.HttpErrorResponse](callback func() (T, error)) (T, error) {
```

Comment:
This looks good, but could you make it part of the client? You could use composition to create a new struct that has all the methods of the *cloudapi.ClientWithResponses, and also has this method.